### DISCUSSION TITLE
How to access c:geo's new integrated Wherigo player?

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#16172] [General]
Somewhere around this [comment](https://github.com/cgeo/cgeo/issues/10087#issuecomment-2394956084) we started a discussion on how to access the cool new Wherigo player integrated in c:geo. I open this discussion here to avoid having it buried in some closed dev issue.

Let's start with collecting use cases for accessing Wherigo player: (feel free to extend this list)
1. I have a link pointing to a cartridge on wherigo.com and would like to play that cartridge
2. I have a cache listing for a Wherigo cache downloaded in c:geo and would like to play that cartridge
3. I have downloaded a cartridge on my phone and would like to play that cartridge
4. I'm running a cartridge currently, but want to do some other caches along the way
5. I'm running a cartridge currently, but am on the map currently - how can I visit my inventory / read the list of tasks / get some other info about currently running WIG?

some ideas:

for 1:
- Catch the uri and start c:geo-internal task to download the cartridge (maybe after some confirmation dialog) - like we do for cache uris

for 2:
- use "Wherigo" icon in cache listing to start the internal player (like we already have); download cartridge if it is not downloaded already
- in the long run the button pointing to WhereYouGo player could be removed (replaced by a button pointing to the integrated player)

for 3:
- as we don't know if there is some cache related to it (so the user can't use variant 2), we need some different means to load the cartridge
- a "Wherigo" in homescreen's overflow menu would be a good place for such management dialog (like we have already)
- additionally, configurable things like quick launch item on home screen or custom icon in bottom navigation are fine as well, but they don't work out of the box (so keep it, but they are no replacement for the menu item)

for 4 & 5:
- when I'm in a WIG currently, tapping on bottom navigation's map button brings me back to the map. The other way around is a bit more difficult:
- when a cartridge is running we have a system notification. Tapping on this opens c:geo's management activity for Wherigos => works fine, only drawback is (IMHO), that this notification is outside of c:geo's window, which may be hard to find. It's ok to have, but IMHO we need more
- proposal: as this is about switching between map (for other caches and/or zones of the WIG) and some additional info about the currently running WIG, why not put a "quick access button" on the map, right beside the "map quick settings" and "route/tracks quick settings" buttons. WIG button will only appear when a WIG is currently running. A bottom dialog could come up with all items relevant for the currently running WIG.

**Comment by eddiemuc** [comment]
@moving-bits thanks for diving into this. I like your approach a lot. Here are my 2 cents:
- for 1: agreed, but let's implement this later once the player is more stable
- for 2: agreed and partly implemented. WhereYouGo button removal should probably wait until player is more stable
- for 3: agreed and implemented 

4&5 are IMHO the core questions: how to access a running wherigo game?

- I agree that notification tap should work but is not enough
- I believe that users who play Wherigos regularly will either add a quick access on main screen or even a custom item in navigation bar. I propose to have a one-time-message suggesting this to users when they open player for the first time
- Wrt "jumping back from WIG to map via bottom navigation": this is only possible if navigation bar is displayed in wig player view. And this is only the case when wherigo is a custom item in the navigation bar. Those users have it easy to jump back and forth :-)
- I like your idea about a quick access button in the map. Also something from the time when player is more stable

**Reply by moving-bits** [reply]
> * for 1: agreed, but let's implement this later once the player is more stable
> * for 2: agreed and partly implemented. WhereYouGo button removal should probably wait until player is more stable

both of it agreed, I mentioned it now just for the sake of completeness

Though for the button we already have for 2 we need to decide whether to show it automatically (like today) or if the user needs to enable some "use internal Wherigo player" switch. For nightly versions it's ok as is, but for a regular release the duplicated button may confuse users.

> 4&5 are IMHO the core questions: how to access a running wherigo game?
> [...]
> * I believe that users who play Wherigos regularly will either add a quick access on main screen or even a custom item in navigation bar.

Probably yes, they just need to know of this feature.

> I propose to have a one-time-message suggesting this to users when they open player for the first time

Might be a way to let them know.

Or we could use an automatism to use the fifth place in bottom navigation for WIG if (and only if) a WIG is running currently (overriding the configured setting for this space temporarily).

> * Wrt "jumping back from WIG to map via bottom navigation": this is only possible if navigation bar is displayed in wig player view. And this is only the case when wherigo is a custom item in the navigation bar. Those users have it easy to jump back and forth :-)

Good point. The support views (like "inventory" etc.) could be just dialogs, not covering the map fully, but the actual game is an activity and (in many cases) needs the screen estate. Here a button in bottom navigation looks like the best solution to me, a good compromise of accessibility and integration in c:geo.

> * I like your idea about a quick access button in the map. Also something from the time when player is more stable

This could be the means to access all support views (inventory, tasks, save/restore current game etc.)

**Comment by fm-sys** [comment]
**Suggestion for 4 & 5**, if no WIG bottom navigation item is configured: 

(show this popup bar in all bottom navigation activities, or maybe even sub-activities as well?)

Clicking the bar would open WIG, the back button would return to c:geo.

![grafik](https://github.com/user-attachments/assets/b78fa596-9739-47b3-b545-e6ca1adca84e)
